ssc/cmod_pvwattsv5.cpp

```diff
@@ -119,575 +119,632 @@ static var_info _cm_vtab_pvwattsv5_part2[] = {
-				}
-
-			}
-
-			// apply hourly shading factors to beam (if none enabled, factors are 1.0)
-			ibeam *= shad_beam;
-
-			// apply sky diffuse shading factor (specified as constant, nominally 1.0 if disabled in UI)
-			iskydiff *= shad_diff;
-
-			poa = ibeam + iskydiff + ignddiff;
-
-			double wspd_corr = wspd < 0 ? 0 : wspd;
-
-			// module cover
-			tpoa = poa;
-			if (aoi > AOI_MIN && aoi < AOI_MAX)
-			{
-				double mod = iam(aoi, use_ar_glass);
-				tpoa = poa - (1.0 - mod)*dni*cosd(aoi);
-				if (tpoa < 0.0) tpoa = 0.0;
-				if (tpoa > poa) tpoa = poa;
-			}
-
-			// cell temperature
-			pvt = (*tccalc)(poa, wspd_corr, tdry);
-
-			// dc power output (Watts)
-			dc = dc_nameplate * (1.0 + gamma * (pvt - 25.0))*tpoa / 1000.0;
-
-			// dc losses
-			dc = dc * (1 - loss_percent / 100);
-
-			// dc degradation
-			dc *= degradationFactor; //this will be 1.0 in year 1 or if degradation isn't specified
-
-			// inverter efficiency
-			double etanom = inv_eff_percent / 100.0;
-			double etaref = 0.9637;
-			double A = -0.0162;
-			double B = -0.0059;
-			double C = 0.9858;
-			double pdc0 = ac_nameplate / etanom;
-			double plr = dc / pdc0;
-			ac = 0;
-
-			if (plr > 0)
-			{ // normal operation
-				double eta = (A*plr + B / plr + C)*etanom / etaref;
-				ac = dc * eta;
-			}
-
-			if (ac > ac_nameplate) // clipping
-				ac = ac_nameplate;
-
-			// make sure no negative AC values (no parasitic nighttime losses calculated)
-			if (ac < 0) ac = 0;
-		}
-		else
-		{
-			poa = 0;
-			tpoa = 0;
-			pvt = tdry;
-			dc = 0;
-			ac = 0;
-		}
-	}
+    }
+
+    void initialize_cell_temp(double ts_hour, double last_tcell = -9999, double last_poa = -9999)
+    {
+        tccalc = new pvwatts_celltemp(inoct + 273.15, PVWATTS_HEIGHT, ts_hour);
+        if (last_tcell > -99 && last_poa >= 0)
+            tccalc->set_last_values(last_tcell, last_poa);
+    }
+
+
+    int process_irradiance(int year, int month, int day, int hour, double minute, double ts_hour,
+        double lat, double lon, double tz, double dn, double df, double alb, double elev, double pres, double tdry)
+    {
+        irrad irr;
+        irr.set_time(year, month, day, hour, minute, ts_hour);
+        irr.set_location(lat, lon, tz);
+        irr.set_optional(elev, pres, tdry);
+        irr.set_sky_model(2, alb);
+        irr.set_beam_diffuse(dn, df);
+        irr.set_surface(track_mode, tilt, azimuth, 45.0,
+            shade_mode_1x == 1, // backtracking mode
+            gcr, false, 0.0);
+
+        int code = irr.calc();
+
+
+        irr.get_sun(&solazi, &solzen, &solalt, 0, 0, 0, &sunup, 0, 0, 0);
+        irr.get_angles(&aoi, &stilt, &sazi, &rot, &btd);
+        irr.get_poa(&ibeam, &iskydiff, &ignddiff, 0, 0, 0);
+
+        return code;
+    }
+
+    int process_irradiance_1ts(int year, int month, int day, int hour, double minute, double ts_hour,
+        double lat, double lon, double tz, double dn, double df, double alb, double elev, double pres, double tdry)
```

Comment:
What is the difference between `process_irradiance` and `process_irradiance_1ts`?

the input parameter `elev`, `pres` and `tdry` in line 251 should be renamed since they shadow fields of cm_pvwattsv5_base (cmod_pvwattsv5.cpp::L131)

Reply:
The fields in cm_pvwattsv5_base  (cmod_pvwattsv5.cpp::L131) were unused so they have been deleted. The different process_irradiance functions I think were from before the optional inputs were added to the single time step module. process_irradiance_1ts has been removed and all uses have been replaced with process_irradiance.